src/error.rs

```diff
@@ -107,7 +107,7 @@ impl core::fmt::Display for Error {
             Error::WorkingCounter { expected, received } => {
                 write!(f, "working counter expected {}, got {}", expected, received)
             }
-            Error::Timeout => f.write_str("timeout"),
+            Error::Timeout(kind) => write!(f, "timeout: {kind}"),
```

Comment:
Nit: for consistency (and my personal preference):

```suggestion
            Error::Timeout(kind) => write!(f, "timeout: {}", kind),
```